### PR TITLE
Support python 3.9

### PR DESCRIPTION
**Pull request by komalali**
Fixes: #5662

### CODE REVIEW

File: sdk/go/common/util/archive/archive.go [hunk 1]
```diff
@@ -76,6 +76,7 @@ func UnTGZ(tarball []byte, dir string) error {
 			return errors.Wrapf(err, "untarring")
 		}
 
+		// nolint:gosec
```
Comment (komalali) [inline sdk/go/common/util/archive/archive.go:79]: Had to add this because gosec was complaining (only in CI, not locally)

**Review by stack72 — APPROVED** [review]
LGTM! Subject to adding the appropriate TO-DO items with regards to supporting Python 3.9 on Windows when the package is available

**Review by lblackstone — APPROVED** [review]

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ jobs:
     strategy:
       matrix:
         go-version: [ 1.15.x ]
-        python-version: [ 3.7.x ]
+        python-version: [ 3.9.x ]
```
Comment (justinvp) [inline .github/workflows/master.yml:28]: Not really a comment for this PR, but hopefully as part of https://github.com/pulumi/pulumi/issues/5664 we can find some way to have coverage of all the versions of Python we support (e.g. 3.6.x, 3.7.x, 3.8.x, and 3.9.x). Maybe as some cron run, and keep the matrix smaller for PRs? cc: @stack72

File: sdk/python/cmd/pulumi-language-python-exec [hunk 3]
```diff
@@ -9,6 +9,17 @@ import sys
 import traceback
 import runpy
 
+# python 3.9 does not have grpcio support on Windows yet [1]. Return an error suggesting a python downgrade in this
+# case. Remove this check once Windows is supported [2].
+# [1] https://github.com/grpc/grpc/issues/24344
+# [2] https://github.com/pulumi/pulumi/issues/5676
```
Comment (justinvp) [inline sdk/python/cmd/pulumi-language-python-exec:15]: Nit: minor rearrangement to make the TODO more clear.

```suggestion
# python 3.9 does not have grpcio support on Windows yet (https://github.com/grpc/grpc/issues/24344).
# Return an error suggesting a python downgrade in this case.
# TODO[pulumi/pulumi#5676]: Remove this check once Windows is supported.
```

**Review by justinvp — APPROVED** [review]
LGTM

File: .github/workflows/master.yml (hunk 2 above)
Comment (komalali) [reply .github/workflows/master.yml:28]: Yeah I agree, I had initially run this as a matrix using `[3.7.x, 3.9.x]` - I think it makes sense to do that as a cron.

**Pull request merged**